apps/backoffice-v2/src/common/hooks/useZodSearchParams/useZodSearchParams.tsx

```diff
@@ -19,13 +19,19 @@ export const useZodSearchParams = <TSchema extends AnyZodObject>(
     [schema, searchParamsAsObject],
   );
   const navigate = useNavigate();
+  const { state } = useLocation();
   const onSetSearchParams = useCallback(
     (searchParams: Record<string, unknown>) => {
       navigate(
         `${pathname}${serializer({
           ...parsedSearchParams,
           ...searchParams,
         })}`,
+        {
+          state: {
```

Comment:
Uses of `navigate` have the potential of wiping `location.state.from`, passing it in again ensures this does not happen.